src/ATen/native/xpu/sycl/MultinomialKernel.cpp

```diff
@@ -328,13 +335,15 @@ struct sampleMultinomialOnceFunctor : public __SYCL_KER_CONFIG_CONVENTION__ {
         if (inBucket) {
           // We're done; we have the sample
           // Torch indices are 1-based
-          atomicMax(
-              sycl_local_ptr<int>(
-                  foundPos_
-                      .template get_multi_ptr<sycl::access::decorated::no>()
-                      .get()),
-              cat);
-          // foundPos_[0] = 1;
+
+          sycl::atomic_ref<
+              int,
+              sycl_mem_odr_rlx,
+              sycl_mem_scp_wg,
+              sycl_local_space>
+              target(foundPos_[0]);
+          target.fetch_max(cat, sycl_mem_odr_acq_rel, sycl_mem_scp_wg);
+
```

Comment:
Keep the API atomicMax. Move fetch_max into the API.
1. >= 32 bits, using fetch_max
2. < 32 bits, using CAS